scripts/sanity_chk/sanitylib.py

```diff
@@ -2163,6 +2169,35 @@ def cleanup_artifacts(self):
                 elif not os.listdir(path):
                     os.rmdir(path)
 
+    def cleanup_device_testing_artifacts(self):
+        logger.debug("Cleaning up for Device Testing {}".format(self.instance.build_dir))
+
+        sanitizelist = [
+            'CMakeCache.txt',
+            'zephyr/runners.yaml',
+        ]
+        keep = [
+            'zephyr/zephyr.hex',
+            'zephyr/zephyr.bin',
+            'zephyr/zephyr.elf',
+            ]
+
+        keep += sanitizelist
+
+        self.cleanup_artifacts(keep)
+
+        # sanitize paths so files are relocatable
+        for file in sanitizelist:
+            file = os.path.join(self.instance.build_dir, file)
+            fin = open(file, "rt")
+            data = fin.read()
+            data = data.replace(canonical_zephyr_base+"/", "")
+            fin.close()
```

Comment:
Wouldn't it be safer to use `open` as a context manager in case of unexpected errors? The marked code can be replaced with:
```
with open(file, "rt") as fin:
    data = fin.read()
    data = data.replace(canonical_zephyr_base + "/", "")
```

Reply:
Done